petab/composite_problem.py

```diff
@@ -0,0 +1,77 @@
+"""PEtab problems consisting of multiple models"""
+import os
+from typing import List, Union, Dict
+
+import pandas as pd
+
+from . import parameters
+from . import problem
+from . import yaml
+
+
+class CompositeProblem:
+    """Representation of a PEtab problem consisting of multiple models
+
+    Attributes:
+        problems:
+            List ``petab.Problems``
+        parameter_df:
+            PEtab parameter DataFrame
+    """
+
+    def __init__(
+            self,
+            parameter_df: pd.DataFrame = None,
+            problems: List[problem.Problem] = None):
+        """Constructor
+
+        Arguments:
+            parameter_df:
+                see CompositeProblem.parameter_df
+            problems:
+                see CompositeProblem.problems
+        """
+        self.problems: List[problem.Problem] = problems
+        self.parameter_df: pd.DataFrame = parameter_df
+
+    @staticmethod
+    def from_yaml(yaml_config: Union[Dict, str]
+                  ) -> 'CompositeProblem':
```

Comment:
this is ugly format. + can one use the class name instead of str for ret type?

Reply:
Not inside the class, because not yet known at that point.